src/compiler/typescript_to_rust/azle_generate/src/generators/canister_methods/query_and_update.rs

```diff
@@ -67,25 +67,28 @@ fn generate_return_expression(fn_decl: &AzleFnDecl) -> proc_macro2::TokenStream
 
     let return_type = fn_decl.get_return_ts_type();
 
-    if type_is_null_or_void(return_type) {
-        return quote! {
-            return;
-        };
-    }
-
-    quote! {
-        _azle_final_return_value.try_from_vm_value(&mut *_azle_boa_context).unwrap()
-    }
-}
-
-/// Returns true if the return type is `null`, or `void`. Otherwise returns false.
-fn type_is_null_or_void(ts_type: &TsType) -> bool {
-    match ts_type {
+    let null_and_void_handler = match return_type {
         TsType::TsKeywordType(keyword) => match keyword.kind {
-            // TODO: Consider handling `TsNeverKeyword` and `TsUndefinedKeyword`
-            TsNullKeyword | TsVoidKeyword => true,
-            _ => false,
+            TsNullKeyword => quote! {
+                if !_azle_final_return_value.is_null() {
+                    ic_cdk::api::trap("TypeError: value is not of type 'null'");
+                }
+            },
+            TsVoidKeyword => quote! {
+                if !_azle_final_return_value.is_undefined() {
+                    ic_cdk::api::trap("TypeError: value is not of type 'void'");
```

Comment:
I think we might want to change this error to describe that undefined must be returned explicitly or implicitly